StreamChatClient_v3/WebSocketClient/Events/UserEvents_Tests.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient_v3
+import XCTest
+
+class UserEvents_Tests: XCTestCase {
+    let userPresenceJSON = XCTestCase.mockData(fromFile: "UserPresence")
+    let userStartWatchingJSON = XCTestCase.mockData(fromFile: "UserStartWatching")
+    let userStopWatchingJSON = XCTestCase.mockData(fromFile: "UserStopWatching")
+    let userBannedJSON = XCTestCase.mockData(fromFile: "UserBanned")
+    let userUnbannedJSON = XCTestCase.mockData(fromFile: "UserUnbanned")
+    
+    let eventDecoder = EventDecoder<DefaultDataTypes>()
+    
+    func test_userPresenceEvent() throws {
+        let event = try! eventDecoder.decode(from: userPresenceJSON) as! UserPresenceChanged<DefaultDataTypes>
+        XCTAssertEqual(event.userId, "steep-moon-9")
+        XCTAssertNotNil(event.createdAt)
+    }
+    
+    func test_watchingEvent() throws {
+        var event = try! eventDecoder.decode(from: userStartWatchingJSON) as! UserWatching<DefaultDataTypes>
+        XCTAssertEqual(event.userId, "broken-waterfall-5")
+        XCTAssertTrue(event.isStarted)
+        XCTAssertTrue(event.watcherCount > 0)
+        XCTAssertEqual(event.cid, ChannelId(type: .messaging, id: "new_channel_7070"))
+        
+        event = try! eventDecoder.decode(from: userStopWatchingJSON) as! UserWatching<DefaultDataTypes>
+        XCTAssertEqual(event.userId, "steep-moon-9")
+        XCTAssertFalse(event.isStarted)
+        XCTAssertTrue(event.watcherCount > 0)
+        XCTAssertEqual(event.cid, ChannelId(type: .messaging, id: "general"))
+    }
+    
+    func test_userBannedEvent() {
+        let event = try! eventDecoder.decode(from: userBannedJSON) as! UserBanned<DefaultDataTypes>
+        XCTAssertEqual(event.userId, "broken-waterfall-5")
+        XCTAssertEqual(event.ownerId, "steep-moon-9")
+        XCTAssertEqual(event.cid, ChannelId(type: .messaging, id: "new_channel_7070"))
+        XCTAssertEqual(event.reason, "I don't like you 🤮")
+    }
+    
+    func test_userUnbannedEvent() {
+        let event = try! eventDecoder.decode(from: userUnbannedJSON) as! UserUnbanned<DefaultDataTypes>
+        XCTAssertEqual(event.userId, "broken-waterfall-5")
+        XCTAssertEqual(event.cid, ChannelId(type: .messaging, id: "new_channel_7070"))
+    }
+}
```

Comment:
nit: In general, I think it's better to add the implementation and tests in the same commit :)